addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/discovery/BigAssFanDiscoveryService.java

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.bigassfan.discovery;
+
+import static org.openhab.binding.bigassfan.BigAssFanBindingConstants.*;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.config.discovery.ExtendedDiscoveryService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BigAssFanDiscoveryService} class implements a service
+ * for discovering the Big Ass Fans.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+public class BigAssFanDiscoveryService extends AbstractDiscoveryService implements ExtendedDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(BigAssFanDiscoveryService.class);
+
+    private static final boolean BACKGROUND_DISCOVERY_ENABLED = true;
+    private final long BACKGROUND_DISCOVERY_DELAY = 8L;
+
+    private DiscoveryServiceCallback callback;
+
+    // Our own thread pool for the long-running listener job
+    private ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> listenerJob;
+
+    DiscoveryListener discoveryListener;
+
+    private boolean terminate;
+
+    private Runnable listenerRunnable = new Runnable() {
+        @Override
+        public void run() {
+            try {
+                listen();
+            } catch (Exception e) {
+                logger.warn("Discovery listener got unexpected exception: {}", e.getMessage());
```

Comment:
Not handling an unexpected exception here causes the job to be canceled silently (i.e. nothing is logged), which is very confusing to the user.   Would you recommend RuntimeException instead of Exception?

I will remove the other unnecessary Exceptions, and will add the stack trace on warnings and errors.